pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -366,14 +385,17 @@ pub mod pallet {
 		pub(crate) fn random_dna(
 			who: &T::AccountId,
 			season: &SeasonOf<T>,
-		) -> Result<Dna, DispatchError> {
+		) -> Result<(Dna, bool), DispatchError> {
 			let dna = (0..season.max_components)
 				.map(|_| {
 					let (random_tier, random_variation) = Self::random_component(who, season);
 					((random_tier << 4) | random_variation) as u8
 				})
 				.collect::<Vec<_>>();
-			Dna::try_from(dna).map_err(|_| Error::<T>::IncorrectDna.into())
+			let is_rare = dna.iter().all(|each| (each >> 4) >= RarityTier::Legendary as u8);
```

Comment:
> We consider an avatar rare if all the tiers of its components are `Legendary` or above.

@DidacSF how does this sound? :arrow_up:

Reply:
Looks good!